src/renderer/src/pass/forward.rs

```diff
@@ -125,13 +168,110 @@ impl<R> Pass<R> for DrawNoShading<R>
     }
 }
 
+gfx_defines!(
+    constant PointLight {
+        propagation: [f32; 4] = "propagation",
+        center: [f32; 4] = "center",
+        color: [f32; 4] = "color",
+    }
+
+    pipeline shaded {
+        vbuf: gfx::VertexBuffer<VertexPosNormal> = (),
+        ka: gfx::Global<[f32; 4]> = "u_Ka",
+        kd: gfx::Global<[f32; 4]> = "u_Kd",
+        lights: gfx::ConstantBuffer<PointLight> = "u_Lights",
+        light_count: gfx::Global<i32> = "u_LightCount",
```

Comment:
I think this should be refactored to avoid globals. It's not like they are here for GL2 compatibility, since you are using a constant buffer for the lights anyway.